src/compiler/scala/tools/nsc/plugins/Plugin.scala

```diff
@@ -60,13 +60,15 @@ abstract class Plugin {
    *  @return true to continue, or false to opt out
    */
   def init(options: List[String], error: String => Unit): Boolean = {
-    if (!options.isEmpty) error(s"Error: $name takes no options")
+    // call to deprecated method required here, we must continue to support
+    // code that subclasses that override `processOptions`.
+    processOptions(options, error
```

Comment:
Sorry, was a push-and-run incident. Not sure if `git revert` got this wrong, or, more likely, I was fat fingered in the process. Just pushed a fix.